src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectOptionsProcessor.cs

```diff
@@ -126,50 +132,58 @@ private void UpdateProjectOptions_NoLock()
 
                 if (effectiveRuleSetPath != null)
                 {
-                    _ruleSetFile = _workspaceServices.GetRequiredService<VisualStudioRuleSetManager>().GetOrCreateRuleSet(effectiveRuleSetPath);
+                    _ruleSetFile = _workspace.Services.GetRequiredService<VisualStudioRuleSetManager>().GetOrCreateRuleSet(effectiveRuleSetPath);
                     _ruleSetFile.Target.Value.UpdatedOnDisk += RuleSetFile_UpdatedOnDisk;
                 }
             }
 
+            if (commandLineArgs == null)
```

Comment:
Why? What about derived types that might not actually get a call to set the command line string?

Reply:
I'll revert the changes to this file. It's too insane to reason about.